src/utils/CopilotAPI.ts

```diff
@@ -159,13 +165,54 @@ export class CopilotAPI {
     )
   }
 
-  async markNotificationAsRead(id: string): Promise<void> {
+  async _markNotificationAsRead(id: string): Promise<void> {
     console.info('CopilotAPI#markNotificationAsRead | token =', this.token)
     await this.copilot.markNotificationRead({ id })
   }
 
-  async deleteNotification(id: string): Promise<void> {
+  async _deleteNotification(id: string): Promise<void> {
     console.info('CopilotAPI#deleteNotification | token =', this.token)
     await this.copilot.deleteNotification({ id })
   }
+
+  // Methods wrapped with retry
+  getTokenPayload = async (...args: unknown[]) => await withRetry(this._getTokenPayload.bind(this), args)
+
+  me = async (...args: unknown[]) => await withRetry(this._me.bind(this), args)
+
+  getWorkspace = async (...args: unknown[]) => await withRetry(this._getWorkspace.bind(this), args)
+
```

Comment:
Instead of writing `withRetry` for each public method, consider using a dynamic approach or decorator to reduce code repetition.

```javascript
wrapWithRetry(method: Function) {
  return async (...args: unknown[]) => await withRetry(method.bind(this), args)
}
constructor(...) {
  this.getTokenPayload = this.wrapWithRetry(this._getTokenPayload)
}
```